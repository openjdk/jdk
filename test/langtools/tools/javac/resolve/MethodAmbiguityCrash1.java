/*
 * @test /nodynamiccopyright/
 * @bug 8337980
 * @summary Test compiler crash due to failure to resolve method ambiguity
 * @compile/fail/ref=MethodAmbiguityCrash1.out -XDrawDiagnostics MethodAmbiguityCrash1.java
 */
public class MethodAmbiguityCrash1 {

    public interface A {
        int op();
    }

    public abstract static class B {
        abstract int op();
    }

    public abstract static class C extends B implements A {

        public static int test() {
            return op();    // compile should fail here
        }
    }
}
