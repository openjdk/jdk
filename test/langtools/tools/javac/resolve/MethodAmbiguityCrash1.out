MethodAmbiguityCrash1.java:20:20: compiler.err.non-static.cant.be.ref: kindname.method, op()
1 error
